doc/keycodes.txt

```diff
@@ -0,0 +1,162 @@
+Minetest Keycodes Reference
+===========================
+
+Key codes used for `minetest.conf` keybindings and returned from formspec
+`key_event` `name` field.
+
+When choosing keys for mods, try not to use arcane keys; instead, choose keys
+that are on every or most keyboards.
+
+* `KEY_LBUTTON`          = Left mouse button
+* `KEY_RBUTTON`          = Right mouse button
```

Comment:
I know this has been brought up before but with the Irrlicht 1.9 switch and possible(?) SDL integration coming I think that making this list of keys essentially part of our API is an awful idea.
Especially the IME keys, `KEY_OEM_2` and arcane keys like `KEY_XBUTTON1` (mouse input shouldn't be here anyway) are likely to go once someone sorts out the input system mess.

I'd solve this differently, specifically I'd identify keys by the character they produce and use reserved names for the rest.
So e.g.:
* printable ASCII space: character keys, always lowercase (example: `a`)
* `<Tab>`: Tab key
* `<Backspace>`: ...
* `<Return>`: ...
* `<Left>`, `<Right>`, `<Up>`, `<Down>`: direction keys

Reply:
Also there's something I don't see explained here: How would Shift+A be sent? Are held down keys sent repeatedly? (I hope not)

Reply:
I don't like it either.  SDL doesn't even provide a certain way to do this: `SDL_GetKeyName` and `SDL_GetScancodeName` can give different results on different platforms (like `SDL_SCANCODE_LGUI` being `"Left GUI"` on Linux and `"Left Windows"` on Windows).

We have our own keycode names used in `KeyCode` for usage in the key menu, but it's meant for human readability.  We could modify that somewhat and base it off of SDL keycodes, only documenting the keys that overlap between SDL and Irrlicht for now, and then documenting the rest of the SDL specific keys later.

Shift is sent as a standalone key (`KEY_L/RSHIFT`), and so is A (`KEY_KEY_A`).  If a key is held down, only the most recently pressed key repeats.  So, holding down Shift and A would look like `KEY_LSHIFT` -> `KEY_KEY_A` -> `KEY_KEY_A` -> `KEY_KEY_A` -> ...  So, you have to manually track shift pressed/releases to detect a Shift+A.

Reply:
Mods shouldn't hardcode the keycodes anyway imo. (I hate it when games hardcode the z and x buttons.)

Testing on my machine showed that Shift isn't sent repeatedly, but A is. It would probably be better to send hold down keys only once.

Reply:
Well, hardcoded keys is necessary.  Perhaps the most useful use-case of this PR is to have keyboard shortcuts, and those must be bound to specific keys.

Key repetition is a useful feature (like in text fields).  I like SDL's implementation, which has a `repeat` field that tells whether or not this current event is a repetition, allowing you to filter it out.

Reply:
I know that the names 
```
* `KEY_LWIN`             = Left Windows key (Microsoft Natural keyboard)
* `KEY_RWIN`             = Right Windows key (Natural keyboard)
```
..are already used in keycode.cpp, but exposing them as such to lua means we can't change the names afterwards as easy. The key has different names on different platforms unix, windows,apple, which are interpreted as the according equivalent anyway:
https://en.wikipedia.org/wiki/Meta_key
https://en.wikipedia.org/wiki/Windows_key
https://en.wikipedia.org/wiki/Command_key

If we're free to call it whatever we want, maybe 'meta key' is more neutral regarding to systems, i.e. equally unknown to both apple users and windows users, but easily guessable.  I know that on kde plasma the name 'meta key' is used, for example.  I didn't find anything on how this is called on android, and I never saw it on android systems' screen keyboards either. Still, given that most keyboards are not apple keyboards, but have a windows logo printed on them, it's also okay to call it that, I guess - especially if we already define the key to this name internally.